vega_sim/scenario/common/agents.py

```diff
@@ -3175,3 +3177,209 @@ def initialise(
             metric=self.metric,
             markets=market_ids,
         )
+
+
+class AtTheTouchMarketMaker(StateAgentWithWallet):
+    """Eventually can use the at-the-touch optimal strategy from
+        Algorithmic and High-Frequency Trading by Cartea, Jaimungal and Penalva.
+
+    Posts only pegged limit orders
+    """
+
+    NAME_BASE = "at_the_touch_market_maker"
+
+    def __init__(
+        self,
+        key_name: str,
+        initial_asset_mint: float = 1000000,
+        market_name: Optional[str] = None,
+        asset_name: Optional[str] = None,
+        market_decimal_places: int = 5,
+        asset_decimal_places: int = 0,
+        position_decimal_places: int = 0,
+        tag: str = "",
+        wallet_name: str = None,
+        peg_offset=1,
+        max_position=1,
+    ):
+        super().__init__(wallet_name=wallet_name, key_name=key_name, tag=tag)
+        self.initial_asset_mint = initial_asset_mint
+        self.mdp = market_decimal_places
+        self.adp = asset_decimal_places
+        self.pdp = position_decimal_places
+
+        self.current_step = 0
+
+        self.market_name = market_name
+        self.asset_name = asset_name
+
+        self.current_position = 0
+        self.order_size = 10 ** (-position_decimal_places)
```

Comment:
Why the tiny order size?

Reply:
Think it was originally a quick work around to not have to consider the partially filled order case - with the changes to how we determine whether an order is live and the max position logic I don't think we need to worry about partially filled orders any more.

@davidsiska-vega as long as you are happy with us not "refreshing" the volume of the order on a partial fill and only doing so once the order is fully filled (of course still following the max position conditions).

Reply:
That may be fine ;-).